corehq/apps/userreports/reports/filters/choice_providers.py

```diff
@@ -394,6 +396,47 @@ def default_value(self, user):
         return None
 
 
+class DomainChoiceProvider(ChainableChoiceProvider):
+
+    def _query_domains(self, domain, query_text):  # should we account for multiple registries?
+        registries = DataRegistry.objects.accessible_to_domain(domain)
```

Comment:
This should only look at the registry linked to the report:
```python
slug = self.report.config.registry_slug
try:
    registry = DataRegistry.objects.accessible_to_domain(domain, slug=registry_slug)
except DataRegistry.DoesNotExist:
    return {domain}
```

Ideally this would use the data registry helper (https://github.com/dimagi/commcare-hq/pull/30021/files#diff-0bfe60b917077cfbec6723904a88dfba3e4235d528dc908c7d5f7a1fb70f64c2R7)

```python
class RegistryReportConfiguration(ReportConfiguration):
    @cached_property
    def registry_helper(self):
        return DataRegistryHelper(self.domain, self.registry_slug)

# then in DomainChoiceProvider:

class DomainChoiceProvider(ChainableChoiceProvider):
    def _query_domains(self, query_text):
        try:
            return self.report.registry_helper.visible_domains
        except RegistryNotFound:
            return {self.domain}
```